controllers/nodehealthcheck_controller.go

```diff
@@ -226,50 +229,38 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return result, nil
 	}
 
-	template, err := resourceManager.GetTemplate(nhc.Spec.RemediationTemplate)
-	if err != nil {
-		log.Error(err, "failed to get template")
-		return result, err
-	}
-
 	// from here on these can change
 	patchInFlightRemediations = true
 
 	// delete remediation CRs for healthy nodes
 	for _, node := range healthyNodes {
-		remediationCR := resourceManager.GenerateRemediationCRBaseNamed(template.GroupVersionKind(), template.GetNamespace(), node.GetName())
-		if deleted, err := resourceManager.DeleteRemediationCR(remediationCR, nhc); err != nil {
-			log.Error(err, "failed to delete remediation CR for healthy node", "node", node.Name)
-			return result, err
-		} else if deleted {
-			log.Info("deleted remediation CR", "name", remediationCR.GetName(), "NHC name", nhc.Name)
-			r.Recorder.Eventf(nhc, eventTypeNormal, eventReasonRemediationRemoved, "Deleted remediation CR for node %s", remediationCR.GetName())
+		remediationCRs := resourceManager.GenerateAllRemediationCRs(&node, nhc)
+		for _, remediationCR := range remediationCRs {
+			if deleted, err := resourceManager.DeleteRemediationCR(remediationCR, nhc); err != nil {
+				log.Error(err, "failed to delete remediation CR for healthy node", "node", node.Name)
+				return result, err
+			} else if deleted {
+				log.Info("deleted remediation CR", "name", remediationCR.GetName(), "NHC name", nhc.Name)
+				r.Recorder.Eventf(nhc, eventTypeNormal, eventReasonRemediationRemoved, "Deleted remediation CR for node %s", remediationCR.GetName())
+			}
 		}
 	}
 
-	minHealthy, err := intstr.GetScaledValueFromIntOrPercent(nhc.Spec.MinHealthy, len(nodes), true)
-	if err != nil {
+	// check if we enough healthy nodes
```

Comment:
Nit: comment phrasing is unclear.
did you mean ` // check if we have enough healthy nodes` ?